pinot-core/src/main/java/org/apache/pinot/core/data/manager/offline/DimensionTableDataManager.java

```diff
@@ -206,8 +206,8 @@ private DimensionTable createFastLookupDimensionTable() {
     // loading is in progress.
     int token = _loadToken.incrementAndGet();
 
-    Schema schema = ZKMetadataProvider.getTableSchema(_propertyStore, _tableNameWithType);
-    Preconditions.checkState(schema != null, "Failed to find schema for dimension table: %s", _tableNameWithType);
+    Schema schema = _indexLoadingConfig.getSchema();
+    assert schema != null;
```

Comment:
nit: Not sure why you changed Preconditions to asserts, but you can conserve error messages by using `assert schema != null : "whatever message that lazyly evaluates " + someExpression;

Reply:
The `IndexLoadingConfig` is cached with schema always set, thus the schema can never be `null`. I usually use `assert` to avoid overhead and just let both IDE and future developer know the field cannot be `null`